examples/auto_scaling_cluster/main.tf

```diff
@@ -0,0 +1,69 @@
+###############################################################################
+# Resource Group
+###############################################################################
+
+module "resource_group" {
+  source = "git::https://github.com/terraform-ibm-modules/terraform-ibm-resource-group.git?ref=v1.0.5"
+  # if an existing resource group is not set (i.e. null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+###############################################################################
+# VPC
+###############################################################################
+
+module "vpc" {
+  source              = "git::https://github.com/terraform-ibm-modules/terraform-ibm-landing-zone-vpc.git?ref=v5.0.1"
+  resource_group_id   = module.resource_group.resource_group_id
+  region              = var.region
+  prefix              = var.prefix
+  tags                = var.resource_tags
+  name                = var.vpc_name
+  address_prefixes    = var.addresses
+  subnets             = var.subnets
+  use_public_gateways = var.public_gateway
+}
+
+###############################################################################
+# Base OCP
+###############################################################################
+locals {
+
+  cluster_vpc_subnets = {
+    default = module.vpc.subnet_detail_map.zone-1
+  }
+  sz_pool = [
+    {
+      subnet_prefix      = "default"
+      pool_name          = "default" # ibm_container_vpc_cluster automatically names default pool "default" (See https://github.com/IBM-Cloud/terraform-provider-ibm/issues/2849)
+      machine_type       = "bx2.4x16"
+      workers_per_zone   = 2
+      minSize            = 1
+      maxSize            = 5
+      enable_autoscaling = false
```

Comment:
Essentially this is the only line that is different, so the whole sz_pool, could be exposed via variables.tf to enable the test to cover both cases without having to duplicate the whole example.

Reply:
I have updated the example to add more worker-pools.